http/http_errors.ts

```diff
@@ -113,6 +114,10 @@ export class HttpError extends Error {
   get expose(): boolean {
     return false;
   }
+  /** The optional headers object that is set on the error. */
+  get headers(): Headers | undefined {
+    return undefined;
+  }
```

Comment:
```suggestion
  abstract get headers(): Headers | undefined;
```

Would like to avoid code that never gets called.

Reply:
Wouldn't we have to add the `abstract` keyword to the class, as well? And I am not sure if you really want to do that because this would require additional changes.

Reply:
> Would like to avoid code that never gets called.

Should we also make `get expose` and `get status` abstract?

Reply:
Actually, upon reflection, there is no need to override this in every descendent.  This value could be stored on `HttpError` as a private field, and the getter can be on `HttpError`.  The value can be picked from the options.

Reply:
The same could actually be done with expose, but that is unrelated to this PR.

Reply:
I can remove the last two commits again if you want that